syncopy/tests/synth_data.py

```diff
@@ -29,26 +30,47 @@ def collect_trials(trial_generator):
     If the underlying trial generating function also accepts
     a `samplerate`, forward this directly.
 
+    If the underlying trial generating function also accepts
+    a `seed`, forward this directly. One can pass a single seed,
+    to be used for all trials, or a list/np.ndarray of seeds with
+    len/size equal to nTrials, with one seed per trial. Leaving
+    at the default value, None, will select a random seed each time,
+    (and it will differ between trials).
+
     The default `nTrials=None` is the identity wrapper and
     just returns the output of the trial generating function
     directly, so a single trial :class:`numpy.ndarray`.
     """
 
     @functools.wraps(trial_generator)
-    def wrapper_synth(nTrials=None, samplerate=1000, **tg_kwargs):
+    def wrapper_synth(nTrials=None, samplerate=1000, seed=None, seed_per_trial=True, **tg_kwargs):
+
+        use_seed_per_trial = False
+        if nTrials is not None and seed is not None and seed_per_trial:  # Use the single seed to create one seed per trial.
+            rng = np.random.default_rng(seed)
+            seed = rng.integers(1000000, size=nTrials)
```

Comment:
wait, so `seed` gets redefined? What you probably mean are the by-trial seeds?!

Edit: Indeed, I think I would prefer to have a new variable ala `trl_seeds`, what do you think?

Reply:
If 'use_seed_per_trial' is true, we want fixed, but different seeds for each trial. If we would use the same seed in each trial for things like white noise, averaging over many white noise channels will NOT remove the noise.

So we need a way to create different but fixed seeds for all the trials. My solution is to generate them from the given seed.

Another option is to have the user pass as many seeds as there are trials. I had implemented this solution first, but it turned out that it leads to a lot of replicated and pointless code when we call these functions. Instead of something like:

    data = generate_whatever(,,, seed=42)

you get:

    data = generate_whatever(,,, seed=np.random.default_rng(seed).integers(1000000, size=nTrials))

in a lot of places.

Reply:
Yes yes, I got this.. generating further by-trial-seeds from an initial seed is a good idea! The only nitpicky thing I had, was the redefinition of the `seed` name, it makes the code harder to read as it potentially hides exactly that sampling strategy (first `seed` is a number, than with `seed_per_trial = True` it is an array).

Reply:
agree, changed in 85300df